static_frame/core/frame.py

```diff
@@ -27,6 +27,7 @@
 from arraykit import resolve_dtype_iter
 from arraykit import split_after_count
 from numpy.ma import MaskedArray
+from pyarrow.lib import ArrowInvalid  # pylint: disable=E0611
```

Comment:
Should this be moved to a dynamic import like L3135?

Reply:
Yes, good catch. `pyarrow` is not a required dependency, so we want to put the import within the function call.